inferno/callbacks.py

```diff
@@ -139,11 +144,85 @@ def on_batch_end(self, net, X, y, train):
         if isinstance(self.scoring, str):  # TODO: make py2.7 compatible
             # scoring is a string
             y = self.target_extractor(y)
-            scorer = getattr(metrics, self.scoring)
+            try:
+                scorer = getattr(metrics, self.scoring)
+            except AttributeError:
+                raise NameError("Metric with name '{}' does not exist, "
+                                "use a valid sklearn metric name."
+                                "".format(self.scoring))
             y_pred = self.pred_extractor(net.module_(to_var(X)))
             score = scorer(y, y_pred)
         else:
             # scoring is a function
             score = self.scoring(net, X, y)
 
         net.history.record_batch(self.name, score)
+
+
+class PrintLog(Callback):
+    def __init__(
+            self,
+            keys=('epoch', 'train_loss', 'valid_loss', 'train_loss_best',
+                  'valid_loss_best', 'dur'),
+            sink=print,
+            tablefmt='simple',
+            floatfmt='.4f',
+    ):
+        self.keys = (keys,) if isinstance(keys, str) else keys
+        self.sink = sink
+        self.tablefmt = tablefmt
+        self.floatfmt = floatfmt
+
+    def initialize(self):
+        self.first_iteration_ = True
+        self.idx_ = {key: i for i, key in enumerate(self.keys)}
+        return self
+
+    def format_row(self, row):
+        row_formatted = []
+        colors = cycle(Ansi)
+
+        for key, item in zip(self.keys, row):
+            if key.endswith('_best'):
+                continue
+
+            if not isinstance(item, Number):
+                row_formatted.append(item)
+                continue
+
+            color = next(colors)
+            # if numeric, there could be a 'best' key
+            idx_best = self.idx_.get(key + '_best')
+
+            is_integer = float(item).is_integer()
+            template = '{}' if is_integer else '{:' + self.floatfmt + '}'
+
+            if (idx_best is not None) and row[idx_best]:
+                template = color.value + template + Ansi.ENDC.value
+            row_formatted.append(template.format(item))
+
+        return row_formatted
+
+    def table(self, data):
+        formatted = [self.format_row(row) for row in data]
+        headers = [key for key in self.keys if not key.endswith('_best')]
+        return tabulate(
+            formatted,
+            headers=headers,
+            tablefmt=self.tablefmt,
+            floatfmt=self.floatfmt,
+        )
+
+    def on_epoch_end(self, net, *args, **kwargs):
+        data = net.history[-1:, self.keys]
```

Comment:
What if the key does not exist in the history? This would raise a `KeyError` here.